Docs: Improvements, add spring-boot.md, fix some mistakes (#264)
avaje/avaje-config: docs/guides/profiles.md

## Activating Profiles

> Activate a profile by setting the `config.profiles` property (note: plural):

Reviewer: Generally you want to point to avaje.profiles over this one

```suggestion
Activate a profile by setting the `avaje.profiles` property:
```

Get the active profile:

```java
String profile = Config.get("config.profiles", "dev");
if (profile.equals("prod")) {
  // Production-specific behavior
}
```

## Test Configuration Auto-Loading vs Profile Activation

> These are two distinct mechanisms — do not confuse them:

Reviewer: ```suggestion
These are two distinct mechanisms, do not confuse them:
```

**`application-test.yaml` auto-loading (no activation needed):**
`src/test/resources/application-test.yaml` is a special hardcoded filename.

> avaje-config loads it automatically whenever it is present on the classpath —
> typically during Maven/Gradle test runs. No `config.profiles=test` or

Reviewer: ```suggestion
avaje-config loads it automatically whenever it is present on the classpath,
typically during Maven/Gradle test runs. No `config.profiles=test` or
```

`avaje.profiles=test` is required.

```
src/test/resources/
└── application-test.yaml   ← loaded automatically, no profile activation needed
```

**Explicit profile activation (activation required):**
All other profile files (`application-dev.yaml`, `application-it.yaml`, etc.)
require explicit activation:

```bash
java -Dconfig.profiles=it myapp.jar   # loads application-it.yaml
```

> **Common mistake:** setting `-Davaje.profiles=test` in your test runner to
> load `application-test.yaml`. This is not needed — the file is auto-loaded

> > unconditionally when present in test resources.

Reviewer: Is this actually a common mistake?
